File: src/main/java/com/kaiosantiago/laudopro/entity/Plan.java
package com.kaiosantiago.laudopro.entity;

import com.kaiosantiago.laudopro.entity.enums.PlanType;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Data
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "plans")
public class Plan{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, unique = true)
    private PlanType type;

    @Column(nullable = false)
    private Double price;
}

File: LaudoPro.Domain/Models/Plan.cs
using LaudoPro.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LaudoPro.Domain.Models
{
    [Table("plans")]
    public class Plan
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [Column("type")]
        public PlanType Type { get; set; }

        [Required]
        public double Price { get; set; }
    }
}
